collect_app/src/androidTest/java/org/odk/collect/android/espressoutils/pages/EditSavedFormPage.java

```diff
@@ -56,13 +57,22 @@ public EditSavedFormPage checkInstanceState(String instanceName, String desiredS
         return this;
     }
 
-    public OkDialog clickSavedFormWithDialog(String instanceName) {
-        onView(withText(instanceName)).perform(click());
+    public OkDialog clickOnFormWithDialog(String instanceName) {
+        clickOnForm(instanceName);
         return new OkDialog(rule).assertOnPage();
     }
 
     public IdentifyUserPromptPage clickOnFormWithIdentityPrompt(String formName) {
-        onView(withText(formName)).perform(click());
+        clickOnFormButton(formName);
         return new IdentifyUserPromptPage(formName, rule).assertOnPage();
     }
+
+    public FormEntryPage clickOnForm(String formName) {
+        clickOnFormButton(formName);
+        return new FormEntryPage(formName, rule);
+    }
+
+    private void clickOnFormButton(String formName) {
```

Comment:
What do you mean by form button here?

Reply:
Yeah "simple button" felt confusing but this isn't actually better. Let me have another go.

Reply:
Yeah that's not a great name. Will rethink.